lib/util/xdp_sample.c

```diff
@@ -1203,13 +1208,17 @@ static int __sample_remove_xdp(int ifindex, __u32 prog_id, int xdp_flags)
 		}
 	}
 
-	return bpf_set_link_xdp_fd(ifindex, -1, xdp_flags);
+	return xdp_program__detach(xdp_prog, ifindex, xdp_flags, 0);
 }
 
-int sample_install_xdp(struct bpf_program *xdp_prog, int ifindex, bool generic,
-		       bool force)
+int sample_load_and_install_xdp(struct bpf_object *obj,
+				struct bpf_program *prog, int ifindex,
+				bool generic)
```

Comment:
This function is way too complicated now; and mostly redundant as well: the caching of prog_id is not longer needed (and won't work the way it's being done now), because libxdp essentially does the same thing inside the xdp_prog objects. So I'd just get rid of this function entirely and let the callers call xdp_program__attach/detach().